Sources/TSCUtility/CMakeLists.txt

```diff
@@ -47,6 +47,13 @@ target_link_libraries(TSCUtility PUBLIC
   TSCBasic)
 target_link_libraries(TSCUtility PRIVATE
   TSCclibc)
+
+if(CMAKE_SYSTEM_NAME STREQUAL Linux OR CMAKE_SYSTEM_NAME STREQUAL Darwin)
+target_link_libraries(TSCUtility PRIVATE
+  dl
```

Comment:
`dl` is not a safe bet - not all platforms require it.  IIRC, bionic doesn't require `libdl` linkage.  `CMAKE_DL_LIBS` is a better option to use here.

`pthread` is also generally not very great.  Using `Threads::Threads` is better via `find_package(Threads)`